### PR TITLE
Added service and in memory implementation 

### PR DESCRIPTION
**Pull request by RaphealOlumofe**
Added service and in memory implementation

### CODE REVIEW

File: domain/build.gradle [hunk 1]
```diff
@@ -4,5 +4,10 @@ dependencies {
 
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.10.3'
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.3'
+    compile group: 'com.h2database', name: 'h2', version: '1.3.148'
+    compile group: 'org.hibernate.javax.persistence', name: 'hibernate-jpa-2.1-api', version: '1.0.0.Final'
+    compile 'org.springframework.boot:spring-boot-starter-web:2.0.1.RELEASE'
```
Comment (petermcneil) [inline domain/build.gradle:9]: Spring and database stuff shouldn't go inside the domain library.

File: frontend/src/test/java/starter/project/frontend/CarControllerTest.java [hunk 2]
```diff
@@ -1,111 +1,7 @@
 package starter.project.frontend;
 
-import org.junit.jupiter.api.Test;
```
Comment (petermcneil) [inline frontend/src/test/java/starter/project/frontend/CarControllerTest.java:3]: why is this test removed?

File: frontend/src/main/java/starter/project/frontend/CarController.java [hunk 3]
```diff
@@ -4,72 +4,62 @@
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.server.ResponseStatusException;
 import starter.project.domain.Car;
+import starter.project.domain.Service;
+
 import java.util.*;
 
 @RestController
 public class CarController {
-   private Map<Integer, Car> map;
```
Comment (petermcneil) [inline frontend/src/main/java/starter/project/frontend/CarController.java:11]: you've completely changed this endpoint

File: domain/build.gradle (hunk 1 above)
Comment (petermcneil) [reply domain/build.gradle:9]: you should move this to the frontend package.

**Review by petermcneil — CHANGES_REQUESTED** [review]
The way we do our databases isn't through Spring typically, but this looks good :) 

A few changes due to changes in the CarController.

File: frontend/src/main/java/starter/project/frontend/CarController.java (hunk 3 above)
Comment (RaphealOlumofe) [reply frontend/src/main/java/starter/project/frontend/CarController.java:11]: i didnt think private Map<Integer, Car> map; still had a purpose as the class wouldn't be using it store the data anymore

File: domain/build.gradle (hunk 1 above)
Comment (RaphealOlumofe) [reply domain/build.gradle:9]: when moved to the frontend package the service does not pick it up. Also alot of the annotations used in the domain package also stop working

**Pull request merged**